### ISSUE TITLE
use_funded_or_actuall_staffing switch not always compatible with rescaling by population size

### ISSUE THREAD

**Issue by marghe-molaro** [UCL/TLOmodel#1395]
"actual" and "funded" assumptions around staffing are based on 2018 data. 
If wishing to expand them with population growth, they would therefore need to start rescaling from 2019 onwards.
This clashes with the option to switch from "actual" to "funded" (or vice versa) at an arbitrary year year_use_funded_or_actuall_staffing_switch. E.g. if deciding to switch in 2024, capabilities in that year would not have benefitted from the population expansion that should have occurred from 2019 onwards, but rather will be the values assumed from the 2018 data + 1 year of population rescaling.

**Comment by tbhallett** [comment]
Good point. So, maybe inside this function .... https://github.com/UCL/TLOmodel/blob/e8c94e43c8e031f5acdaa7c4f338552287757f4a/src/tlo/methods/healthsystem.py#L934-L944

.... we need to also effectively do any scaling that has been called from within `RescaleHRCapabilities_ByDistrict` and `ConstantRescalingHRCapabilities` and `DynamicRescalingHRCapabilities`.

One way that I occurs to me or doing this would be:
* When those functions recursively develop the scaling-factors (on minutes of time per officer and per district), they're saved to a "scaling_matrix" rather than directly operating on `self._daily_capabilities`.
* When we call `self._daily_capabilities`, what we want returned is actually `self._original_daily_capabilities * scaling_matrix` (where `self._original_daily_capabilities` is the original version of scaled/funded capabilities)
* This way, we can swap in different values for `self._original_daily_capabilities` and still the scaling that has "already happened" will update them.

There are probably a million other ways  though, and you've probably also already solved it.

**Comment by marghe-molaro** [comment]
Yes there's quite a possible few solutions. Notice that in the case of scaling with population growth this still wouldn't solve the issue of having to rescale the 'new' post-switch staffing by a factor of pop_in_year_of_switch/pop_in_2018 - i.e. we'd have to store the pop in 2018 somewhere.

I think for me there's a more fundamental issue with mixing the switch in staffing and the rescaling: the latter (in all its forms, whether as a one-off or dynamic) was designed to be a _proportional_ rescaling of initially assumed resources, which ensures it is always safe to take place at any place and in no matter what order. But in this case we want to change the initially assumed resources halfway, even if those were being potentially rescaled before that switch. Even conceptually this is a bit odd: what do we mean by switching from actual to funded if the actual resources were already been scaled e.g. by GDP growth? What if by the time we invoke the staff switch they had already been rescaled to an amount greater than funded?

**Comment by marghe-molaro** [comment]
Responsibility for ensuring consistency between expansion assumptions in different scenarios arguably rests with the user, as it would be impractical to implement warnings for every possible combination of expansion options. On that basis, I think we can close this issue.
